packages/@icon-magic/cli/src/index.ts

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+import * as iconBuild from '@icon-magic/build';
+import * as configReader from '@icon-magic/config-reader';
+import * as iconGenerate from '@icon-magic/generate';
+import * as program from 'commander';
+
+import { printUsage } from './usage';
+
+program.command('build [inputPaths...]').action(async inputPaths => {
+  if (!inputPaths.length) {
+    console.log(
+      "No Input Directories were specified.\nDid you mean 'icon-magic build .'?"
+    );
+  }
+  try {
+    // Get the iconSet from the inputPaths
+    const iconSet = configReader.getIconConfigSet(inputPaths);
+
+    // build all the icons
+    await iconBuild.build(iconSet);
+
+    // exit without any errors
+    process.exit(0);
+  } catch (e) {
+    // TODO: display errors here
+    process.exit(1);
+  }
+});
+
+program.command('generate [inputPaths...]').action(async inputPaths => {
+  if (!inputPaths.length) {
+    console.log(
+      "No Input Directories were specified.\nDid you mean 'icon-magic generate .'?"
+    );
+  }
+  try {
+    // Get the iconSet from the inputPaths
+    const iconSet = configReader.getIconConfigSet(inputPaths);
+
+    // generate all the icons
+    await iconGenerate.generate(iconSet);
+
+    // exit without any errors
+    process.exit(0);
+  } catch (e) {
+    // TODO: display errors here
+    process.exit(1);
+  }
+});
+
+program
+  .command('distribute')
+  .option('-i, --inputPaths', 'path to input directory')
+  .option('-o, --outputPath', 'path to output directory')
+  .action(() => {
+    console.log('Calling distribute ... ');
+    // TODO: fill this when it's ready
+    process.exit(0);
+  });
+
+// for all other commands or rather, no command and only arguments
+program.command('* [inputPaths...]').action(async inputPaths => {
```

Comment:
To be safe, the CLI should throw if we get a command that we don't recognize... could be a typo.

It looks like Commander has an option for define a default command: https://www.npmjs.com/package/commander#git-style-sub-commands

Reply:
As we discussed, this is what will work for `icon-magic <inputPaths>` (without any command) and will treat all the params as inputPaths. Also, the default option works only for github style commands (description is passed and no .action is specified against the command)